docs/source/tutorials_source/package/tutorial_simsiam_esa.py

```diff
@@ -0,0 +1,447 @@
+# vertical flips as well as random rotation transformations. We apply weak color 
+# jitter to learn an invariance of the model with respect to slight changes in
+# the color of the water.
+#
+
+# the collate function defines the transformations for self-supervised learning
+collate_fn = lightly.data.ImageCollateFunction(
+    input_size=input_size,
+    # require invariance to flips and rotations
+    hf_prob=0.5,
+    vf_prob=0.5,
+    rr_prob=0.5,
+    # use a weak color jitter for invariance w.r.t small color changes
+    cj_prob=0.5,
+    cj_bright=0.1,
+    cj_contrast=0.1,
+    cj_hue=0.0,
+    cj_sat=0.1,
+)
+
+# create a torchvision transformation for embedding the dataset after training
+test_transforms = torchvision.transforms.Compose([
+    torchvision.transforms.Resize((input_size, input_size)),
+    torchvision.transforms.ToTensor(),
+    torchvision.transforms.Normalize(
+        mean=lightly.data.collate.imagenet_normalize['mean'],
+        std=lightly.data.collate.imagenet_normalize['std'],
+    )
+])
+
+# create a lightly dataset for training (no transforms!)
+dataset_train_simsiam = lightly.data.LightlyDataset(
+    input_dir=path_to_data
+)
+
+# create a lightly dataset for embedding
+dataset_test = lightly.data.LightlyDataset(
+    input_dir=path_to_data,
+    transform=test_transforms
+)
+
+# create a dataloader for training
+dataloader_train_simsiam = torch.utils.data.DataLoader(
+    dataset_train_simsiam,
+    batch_size=batch_size,
+    shuffle=True,
+    collate_fn=collate_fn,
+    drop_last=False,
+    num_workers=num_workers
+)
+
+# create a dataloader for embedding
+dataloader_test = torch.utils.data.DataLoader(
+    dataset_test,
+    batch_size=batch_size,
+    shuffle=False,
+    drop_last=False,
+    num_workers=num_workers
+)
+
+# %%
+# Create the SimSiam model
+# -----------------------
+#
+# Create a ResNet backbone and remove the classification head
+
+# we use a pretrained resnet for this tutorial to speed
+# up training time but you can also train one from scratch
+resnet = torchvision.models.resnet18()
+backbone = nn.Sequential(*list(resnet.children())[:-1])
+
+# create the SimSiam model using the backbone from above
+model = lightly.models.SimSiam(
+    backbone,
+    num_ftrs=num_ftrs,
+    proj_hidden_dim=pred_hidden_dim,
+    pred_hidden_dim=pred_hidden_dim,
+    out_dim=out_dim,
+    num_mlp_layers=num_mlp_layers
+)
+
+# %%
+# SimSiam uses a symmetric negative cosine similarity loss and does therefore
+# not require any negative samples. We build a criterion and an optimizer.
+
+# SimSiam uses a symmetric negative cosine similarity loss
+criterion = lightly.loss.SymNegCosineSimilarityLoss()
+
+# scale the learning rate 
+lr = 0.05 * batch_size / 256
+# use SGD with momentum and weight decay
+optimizer = torch.optim.SGD(
+    model.parameters(),
+    lr=lr,
+    momentum=0.9,
+    weight_decay=5e-4
+)
+
+
+# %% 
+# Use a GPU if available
+
+
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
```

Comment:
Perhaps put this block in line 208 just before the 'model.to(device)' as they belong together.